tests/basic/shared_nfsv4_test.go

```diff
@@ -0,0 +1,173 @@
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nfsserverfailover-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+
+		for _, ctx := range contexts {
+			oneMinute := time.Second * time.Duration(60)
+			var nodesNonReplica []node.Node
+			var volume *volume.Volume
+			Step("disable scheduling on non replica nodes", func() {
+				vols, err := Inst().S.GetVolumes(ctx)
+				if err != nil {
+					logrus.Infof("get volumes error %v", err)
+				}
+				volume = vols[0]
+				replicaSets, err := Inst().V.GetReplicaSets(volume)
+				if err != nil {
+					logrus.Infof("get replicaSets error %v", err)
+				}
+				var replicaNodes []string
+				for _, replicaSet := range replicaSets {
+					nodes := replicaSet.Nodes
+					replicaNodes = append(replicaNodes, nodes...)
+				}
+				allNodes := node.GetWorkerNodes()
+				for _, node := range allNodes {
+					if !contains(replicaNodes, node.VolDriverNodeID) {
+						nodesNonReplica = append(nodesNonReplica, node)
+					}
+				}
+				for _, node := range nodesNonReplica {
+					Inst().S.DisableSchedulingOnNode(node)
+				}
+
+			})
+
+			// scale down and then scale up the app, so that pods are only scheduled on replica nodes
+			Step(fmt.Sprintf("scale down app: %s to 0 ", ctx.App.Key), func() {
+				applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for name := range applicationScaleUpMap {
+					applicationScaleUpMap[name] = int32(0)
+				}
+				err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			Step(fmt.Sprintf("scale up app: %s to 2, and re-enable scheduling on all nodes", ctx.App.Key), func() {
+				applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for name := range applicationScaleUpMap {
+					applicationScaleUpMap[name] = int32(2)
+				}
+				err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+				Expect(err).NotTo(HaveOccurred())
+				ValidateApplications(contexts)
+				for _, node := range nodesNonReplica {
+					Inst().S.EnableSchedulingOnNode(node)
+				}
+			})
+
+			Step("fail over nfs server, and make sure the pod on server gets restarted", func() {
+				oldServer, err := Inst().V.GetNodeForVolume(volume, defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("old nfs server %v [%v]", oldServer.SchedulerNodeName, oldServer.Addresses[0])
+				pods, err := core.Instance().GetPodsUsingPV(volume.ID)
+				Expect(err).NotTo(HaveOccurred())
+				var oldPodOnOldServer corev1.Pod
+				for _, pod := range pods {
+					if pod.Spec.NodeName == oldServer.Name {
+						oldPodOnOldServer = pod
+					}
+				}
+				// make sure there is a pod running on the old nfs server
+				Expect(&oldPodOnOldServer).NotTo(BeNil())
```

Comment:
This check is bogus? It will always succeed since `oldPodOnOldServer` is a local variable. Its address is going to be non-nil.